synapse/handlers/pagination.py

```diff
@@ -399,23 +435,49 @@ def get_delete_ids_by_room(self, room_id: str) -> Optional[StrCollection]:
         """
         return self._delete_by_room.get(room_id)
 
```

Comment:
I think we need to update these functions to pull the delete status from the `rooms_to_purge` database table instead of `self._delete_by_id` which only holds onto the things this instance has worked on its life

Reply:
This is solved in a nice way (I hope ^^) with my last commit. Everything is using a DB, including the purge history actions.
Sorry for the size of this new commit however :grimacing: